symengine/symengine_casts.h

```diff
@@ -87,6 +90,51 @@ inline To down_cast(From &f)
     return *static_cast<ToAsPointer>(&f);
 }
 
+template <typename To, typename From>
+inline To
+numeric_cast(From f,
+             typename std::enable_if<(std::is_signed<From>::value
+                                      && std::is_signed<To>::value)
+                                     || (std::is_unsigned<From>::value
+                                         && std::is_unsigned<To>::value)>::type
+                 * = nullptr)
+{
+    SYMENGINE_ASSERT(f <= std::numeric_limits<To>::max());
+    SYMENGINE_ASSERT(f >= std::numeric_limits<To>::min());
+    return static_cast<To>(f);
```

Comment:
@certik, in Release mode `numeric_cast` should have the same cost as `static_cast` (which is zero)